### PR TITLE
rusk: Add gRPC service for Network communication

### LINKED ISSUE #380: rusk: Add gRPC service for Network communication
**Describe what you want implemented** 
The Rusk server must expose a gRPC service to handle network communication. Through this service, a caller will be able to send messages to the network and listen for incoming messages

**Describe "Why" this is needed**
Considering that:
- we want to integrate the node network layer with the new rust kadcast library, a gRPC seems the best choice to let communicate these 2 modules
- the node is already integrated with other gRPC services exposed by rusk
- other services will require to communicate with rusk and the network layer too

Said that, the inclusion of the Network gRPC service inside Rusk seems the best overall choice

**Describe alternatives you've considered**
Implement the network gRPC service as an isolated executable

**Additional context**
N/A

### PR DESCRIPTION
**Pull request by herr-seppia**
Resolves #380

### CODE REVIEW

File: rusk/src/bin/main.rs [hunk 1]
```diff
@@ -73,8 +76,9 @@ async fn main() {
                 .default_value("info")
                 .help("Output log level")
                 .takes_value(true),
-        )
-        .get_matches();
+        );
+    let app = self::network_config(app);
```
Comment (vlopes11) [inline rusk/src/bin/main.rs:80]: `self` is a bit confusing here.

```suggestion
    let app = network_config(app);
```

File: rusk/src/bin/main.rs [hunk 2]
```diff
@@ -181,7 +195,53 @@ async fn startup_with_tcp_ip(
...
+        .serve(addr)
+        .await?)
+}
+
+/// Setup clap to handle kadcast network configuration
+fn network_config<'a>(app: App<'a, 'a>) -> App<'a, 'a> {
+    app.arg(
+        Arg::with_name("kadcast_public_address")
+            .long("kadcast_public_address")
+            .help("Public address you want to be identified with. Eg: 193.xxx.xxx.198:9999")
```
Comment (vlopes11) [inline rusk/src/bin/main.rs:213]: Can we put in the description the difference between the public address and the listen address?

File: rusk/src/lib/services/network.rs [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+    sender: Sender<(Vec<u8>, SocketAddr, u8)>,
+}
+
+impl RuskNetwork {
+    pub fn new(
+        public_addr: String,
+        listen_addr: Option<String>,
+        bootstrap: Vec<String>,
+    ) -> RuskNetwork {
+        let grpc_sender = broadcast::channel(100).0;
```
Comment (vlopes11) [inline rusk/src/lib/services/network.rs:32]: We create a channel, but discard the receiver. Can we put a comment here on why that isn't a problem?

File: rusk/src/lib/services/network.rs [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+    }
+}
+
+#[tonic::async_trait]
+impl Network for RuskNetwork {
+    async fn send(
+        &self,
+        request: Request<SendMessage>,
+    ) -> Result<Response<Null>, Status> {
+        info!("Recieved SendMessage request");
```
Comment (vlopes11) [inline rusk/src/lib/services/network.rs:73]: This looks like `trace` instead of `info`

**Review by vlopes11 — COMMENTED** [review]
Couple if nits

**Review by vlopes11 — CHANGES_REQUESTED** [review]
LGTM overall, but need to check the problem with dusk analyzer (CI job) and rebase to current `testnet`

**Comment by herr-seppia** [comment]
> LGTM overall, but need to check the problem with dusk analyzer (CI job) and rebase to current `testnet`

Anything should be OK now

**Review by vlopes11 — APPROVED** [review]
LGTM

**Pull request merged**